make/migrations/postgresql/0030_2.0.0_schema.up.sql

```diff
@@ -184,3 +184,43 @@ CREATE TABLE audit_log
  username       varchar(255) NOT NULL,
  op_time        timestamp default CURRENT_TIMESTAMP
 );
+
+/*migrate access log to audit log*/
+/*TODO drop table access_log?*/
+DO $$
+DECLARE
+    access RECORD;
+BEGIN
+    FOR access IN SELECT * FROM access_log
+    LOOP
+      /*insert project create and delete*/
+      IF access.operation = 'create' OR access.operation = 'delete' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, access.operation, 'project', access.repo_name, access.username, access.op_time);
+      END IF;
+      /*insert push image -> create artifact*/
+      IF access.operation = 'push' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, 'create', 'artifact', CONCAT(access.repo_name,'/',access.repo_tag), access.username, access.op_time);
+      END IF;
+      /*insert pull image -> pull tag*/
+      IF access.operation = 'pull' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, 'pull', 'tag', CONCAT(access.repo_name,'/',access.repo_tag), access.username, access.op_time);
+      END IF;
+    END LOOP;
+END $$;
+
+/*migrate access log to audit log*/
+/*user the first push operation to generate the create repository log*/
+DO $$
```

Comment:
If we generate the audit log for repository creation, how about the repository deletion? 
We cannot handle the case that the image is pushed, then deleted, then pushed again, then deleted, which means the repository is created, then deleted and created, deleted

Reply:
that's why it just handles the first creation

Reply:
removed.